### DISCUSSION TITLE
Help montaging multiple images centered

### DISCUSSION THREAD

**Discussion by norberthes** [ImageMagick/ImageMagick#2266] [Help]
Hello, 

I want to montage multiple images into one grid.  When the catalogue has e.g. 10 images and I use this command:

`montage *.jpg  -geometry 16x16+1+1  output.jpg` 

I get this result:

![image](https://user-images.githubusercontent.com/65822467/87476397-9ccda080-c626-11ea-866f-3d62b1aa8dc4.png)

How should I do it to center the last row when it happens to not have enough pictures to form a full row? Desired output:

![image](https://user-images.githubusercontent.com/65822467/87476679-15346180-c627-11ea-9b07-f85eda4d40fd.png)

I don't want to use :null to fill the gaps and force it manually, because the number of images will differ everytime I start the script.

**Comment by fmw42** [comment]
I do not think that is possible with montage. You either need to use append in rows and then append the rows together to stack them. Or you need to compute how many full rows you will have and then how many images in the last row and put in the null: images as needed.

**Comment by snibgo** [comment]
If you know how many rows montage makes, you could crop into rows, "-trim", then "-append" vertically (with "-gravity center").

But I don't know how you would easily find the number of rows.

**Reply by fmw42** [reply]
Isn't the number of rows simply the number of images divided by the number of num_columns set in the -tile argument of montage? That may be a float (non-integer). So the number of full rows in num_full_rows=floor(number_images/num_columns). The number of remaining images in the last row will be num_last_row_images=number_images-(num_full_rows*num_columns), if number_images/num_columns is not a whole number (i.e. has no fractional part).
